src/ride/controllers/search_ride_controller.py

```diff
@@ -0,0 +1,49 @@
+import datetime
+
+from api.utils import success_response
+
+from ..models import Path
+from ..models import Ride
+
+
+class SearchRideController:
+    def __init__(self, data, request, serializer):
+        self._request = request
+        self._data = data
+        self._serializer = serializer
+
+    def process(self):
+        departure_datetime = self._data.get("departure_datetime")
+        start_location_name = self._data.get("start_location_name")
+        end_location_name = self._data.get("end_location_name")
+        if not (
+            Path.objects.filter(
+                start_location_name=start_location_name,
+                end_location_name=end_location_name,
+            )
+        ).exists():
+            return success_response([])
+
+        splitted_list = departure_datetime.split("-")
```

Comment:
instead of splitting, just do
`departure_datetime_object = datetime.datetime.fromisoformat(departure_datetime)`
https://docs.python.org/3/library/datetime.html#datetime.datetime.fromisoformat

Reply:
+1

Reply:
?